src/core/Table/Table.tsx

```diff
@@ -478,22 +551,39 @@ export const Table = <
             });
             return (
               <div {...headerGroupProps} key={headerGroupProps.key}>
-                {headerGroup.headers.map((column) => {
+                {headerGroup.headers.map((column, index) => {
+                  const {
+                    onClick: onSortClick,
+                    ...sortByProps
+                  } = column.getSortByToggleProps() as {
+                    onClick:
+                      | React.MouseEventHandler<HTMLDivElement>
+                      | undefined;
+                    style: React.CSSProperties;
+                    title: string;
+                  };
                   const columnProps = column.getHeaderProps({
-                    ...column.getSortByToggleProps(),
+                    ...sortByProps,
                     className: cx(
                       'iui-cell',
                       { 'iui-actionable': column.canSort },
                       { 'iui-sorted': column.isSorted },
                       column.columnClassName,
                     ),
-                    style: { ...getCellStyle(column) },
+                    style: { ...getCellStyle(column, state.isTableResizing) },
                   });
                   return (
                     <div
                       {...columnProps}
                       key={columnProps.key}
                       title={undefined}
+                      ref={(el) => {
+                        if (el) {
+                          columnRefs.current[column.id] = el;
```

Comment:
I guess we dont need to do that if resize is disabled?

Reply:
Added